packages/nextjs/test/utils/withSentry.test.ts

```diff
@@ -0,0 +1,92 @@
+import * as Sentry from '@sentry/node';
+import * as utils from '@sentry/utils';
+import { NextApiHandler, NextApiRequest, NextApiResponse } from 'next';
+
+import { AugmentedNextApiResponse, withSentry, WrappedNextApiHandler } from '../../src/utils/withSentry';
+
+const FLUSH_DURATION = 200;
+
+async function sleep(ms: number): Promise<void> {
+  await new Promise(resolve => setTimeout(resolve, ms));
+}
+
+async function callWrappedHandler(wrappedHandler: WrappedNextApiHandler, req: NextApiRequest, res: NextApiResponse) {
+  await wrappedHandler(req, res);
+
+  // Within the wrapped handler, we await `flush()` inside `res.end()`, but nothing awaits `res.end()`, because in its
+  // original version, it's sync (unlike the function we wrap around it). The original does actually *act* like an async
+  // function being awaited - subsequent steps in the request/response lifecycle don't happen until it emits a
+  // `prefinished` event (which is why it's safe for us to make the switch). But here in tests, there's nothing past
+  // `res.end()`, so we have to manually wait for it to be done.
+  await sleep(FLUSH_DURATION);
+  while (!res.finished) {
+    await sleep(100);
+  }
```

Comment:
You're right - if we break something in the future (fail to call `res.end()`, for example), running this will cause an infinite loop. Isn't that it's own kind of failure, though? That said, reading over the comment, I do think this could be clearer. I'll take another stab at it.

Reply:
I get that it looks like it, but it's not actually, though - or at least not unless we break the tests. Consider the two mock handlers:

- Non-erroring handler: This calls `res.send()`, which calls `res.end()`, which sets `finished` to true, which stops the loop (or prevents it from starting in the first place).

- Erroring handler: This throws at the `await wrappedHanlder(req, res)` line above, so we never even reach the loop.

I'm not sure what you mean about it finishing too early or too late.

Reply:
Reworked the comments both for clarity and to make it clear that the loop finishes.

Reply:
I do think this is overkill, honestly. If the tests break, it's up to the person working on them to fix them. If I were to mess with the mocks or introduce new mocks and get stuck in an infinite loop, I'd have to debug that, but IMHO it's not something worth putting in safeguards for (and making the tests at least slightly harder to read and harder to reason about) since we're really only talking about the mocks here (not the actual code they're testing).